### ISSUE TITLE
[wish] Delete bookmark after opened?

### ISSUE THREAD

**Issue by Snake883** [jingyu9575/tabs-to-bookmark-groups#15]
What's the chances of having a feature that when a bookmark is opened the addon can also delete the bookmark?

I want to temporary store tabs for later use, and close the tabs.
Later, I want to reopen the tab (via saved bookmark).
Now I don't need the saved bookmark (remember, the bookmark was temporary).
I want the bookmark deleted after the tab is opened.

This really useful for when I'm getting a bunch of website, tabs, youtube videos ready to read/watch later.  I don't want to keep the tabs open because: tabs consume a lot of resources, I have a lot tabs, and I don't know when I'll read/watch that tab.  

I current use OneTab, and I have A LOT of temporary/saved tabs.  I noticed OneTab connect to Google, and don't exactly know what its doing with my info.  I can only guess its using my saved tabs/links to profile me.

**Comment by EstherMoellman** [comment]
Hi @snakyjake1 ,

I'm not the Dev, but perhaps my own experience can help you:

At TabToBookmarkGroups add-on when a tab is manually closed, the bookmark automatically is deleted. Therefore, at this add-on a bookmark always will be automatically temporary as long as the tab exists (is not manually closed).
But in the other hand, you don't want to keep thousand of tabs. So, your question comes: How to have both? Temporary bookmark + Closing tabs? The answer is: Groups.

For example, you start your day by opening 50 tabs about economy. But you don't want to read them now. So you create a Group (let's name it "Economy"). And voilà, now you can close or change Economy Group to another group, the 50 tabs and Economy Group are going to be closed but are not going to be deleted (the 50 tabs are going to be closed until you read them later). And later, when you manually start to close one by one these 50 tabs, bookmarks automatically are going to be deleted one by one. After you finished reading the 50 tabs at Economy Group, and the 50 tabs are going to be manually closed by you... the 50 bookmarks automatically are going to be deleted.

In brief, this is the mechanism:
1) Create a tab => a bookmark automatically will be created.
2) Manually close a tab => a bookmark automatically will be deleted.
3) Create a group => create a tab => a bookmark automatically will be created.
4) Close or change the group => the tab will be closed, but won't be deleted => the same goes for the bookmark. As long as you don't manually close a tab, nothing will be deleted, everything automatically will be saved.
5) When you manually close a tab or delete a group => bookmarks automatically will be deleted.

Hope this will help you.

PS: TabToBookmarkGroups is not connecting to anything, it's not collecting/sending data, and it's an open source. Also at this add-on, when a Group is closed, all its tabs inside are closed, so zero tab connections (great for privacy, security and browser performance). With this add-on you can have thousand of  tabs distributed inside "n" groups, that nothing will be connected, and RAM/CPU won't be consumed (until you open tabs one by one).

**Comment by jingyu9575** [comment]
@snakyjake1 There are 2 major differences between this extension and OneTab:
1. Bookmarks are not deleted after loading;
2. When a group is saved, the window is associated to the group, and any more changes will overwrite the group when switching to another one. 

So I think the new mode should implement:
1. Delete bookmarks after loading;
2. No associations between windows and groups. If a window is saved all tabs are closed.

Is this what you want?

**Comment by jingyu9575** [comment]
> OneTab connect to Google

I think Google has a favicon service, and some extensions use it to get websites' favicons. I'm not sure since OneTab is a closed source extension.

**Comment by Snake883** [comment]
> So I think the new mode should implement:
>     1. Delete bookmarks after loading;

Yes, that sounds about what I'm wishing for.  As a workaround, I open the saved bookmark in Firefox, then click Firefox "star" button to unsave it.

The user scenario for this is that these are temporary bookmarks, and looking for an efficient way to remove them after I have opened them back up.  Otherwise my temporary bookmarks will keep accumulating.
